addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -986,6 +996,32 @@ def mark_as_todo(self):
 
         return
 
+    def reserve_pallet(self, pallet_name):
+        """
+        Reserves a pallet for use in a batch.
+
+        Only one pallet can be reserved per batch. The pallet is automatically
+        considered unreserved when another pallet is reserved or the batch is
+        done.
+
+        Raises a ValidationError if the pallet is already reserved for another
+        batch.
+        """
+        PickingBatch = self.env['stock.picking.batch']
+
+        self.ensure_one()
+
+        conflicting_batch = PickingBatch.search([
+            ('u_last_reserved_pallet_name', '=', pallet_name),
+            ('state', '=', 'in_progress'),
+        ])
```

Comment:
You can add and id != self.id to avoid to find self.

Reply:
Fixed!